lib/benchee/profile.ex

```diff
@@ -0,0 +1,98 @@
+defmodule Benchee.Profile do
+  alias Benchee.Output.ProfilePrinter, as: Printer
+  alias Benchee.Suite
+
+  defmodule Benchee.UnknownProfilerError do
+    defexception message: "error"
+  end
+
+  @moduledoc """
+  Profiles each scenario after benchmarking them if the `profile_after` option is either set to:
+    * `true`,
+    * a valid `profiler`,
+    * a tuple of a valid `profiler` and a list of options to pass to it, e.g., `{:fprof, [sort: :own]}`.
+
+  The profiler that will be used is either the one set by the `profiler_after` option or, if set to `true`,
+  the default one (`:cprof`). It accepts however the following profilers:
+    * `:cprof` will profile with `Mix.Task.Profile.Cprof`. It provides information related to the
+    number of function calls.
+    * `:eprof` will profile with `Mix.Task.Profile.Eprof`. It provides information related to the
+    time spent on each function in regard to the total execution time.
+    * `:fprof` will profile with `Mix.Task.Profile.Fprof`. It provides information related to the
+    time spent on each function, both the *total* time spent on it and the time spent on it,
+    *excluding* the time of called functions.
+  """
+
+  @doc """
+  Runs for each scenario found in the suite the `profile/2` function from the given profiler.
+  """
+  @default_profiler :cprof
+  @spec profile(Suite.t()) :: Suite.t()
```

Comment:
Btw. very good job of picking up our "way of coding" - aka speccing the public functions but mostly not the private ones!